### PR TITLE
fix: client schema usage, update wgc router compose to use client schema

### PR DESCRIPTION
**Pull request by devsergiy**
## TODO

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

File: router/core/executor.go [hunk 1]
```diff
@@ -72,32 +72,43 @@ func (b *ExecutorConfigurationBuilder) Build(ctx context.Context, routerConfig *
...
+		if report.HasErrors() {
+			return nil, fmt.Errorf("failed to parse graphql client schema from engine config: %w", report)
+		}
+		err = asttransform.MergeDefinitionWithBaseSchema(&clientSchema)
+		if err != nil {
+			return nil, fmt.Errorf("failed to merge graphql client schema with base schema: %w", err)
+		}
+		clientSchemaDefinition = &clientSchema
+	} else {
+		clientSchemaDefinition = &routerSchemaDefinition
```
Comment (jensneuse) [inline router/core/executor.go:103]: This is a backwards compatible fallback, right? Can we mark it as such and put a date for removal?

File: router/core/operation_planner.go [hunk 2]
```diff
@@ -60,8 +61,8 @@ func (p *OperationPlanner) preparePlan(requestOperationName []byte, requestOpera
 
 	validation := astvalidation.DefaultOperationValidator()
 
-	// validate the document before planning
-	state := validation.Validate(&doc, p.executor.Definition, &report)
+	// validate the document against client schema before planning
+	state := validation.Validate(&doc, p.executor.ClientSchema, &report)
```
Comment (jensneuse) [inline router/core/operation_planner.go:65]: Are we normalizing, etc... against the client schema as well? It would sound reasonable to me.

**Review by jensneuse — CHANGES_REQUESTED** [review]

**Review by StarpTech — APPROVED** [review]
LGTM

File: router/core/operation_planner.go (hunk 2 above)
Comment (devsergiy) [reply router/core/operation_planner.go:65]: yeah, everything except planning should be done against client schema

File: router/core/executor.go (hunk 1 above)
Comment (devsergiy) [reply router/core/executor.go:103]: yes

What should be a timeframe for this?

Should we log warning? e.g. "you are using outdated wgc version"

Comment (StarpTech) [reply router/core/executor.go:103]: No, the ClientSchema is optional and only used with schema contracts. If schema contracts aren't used the router schema is the client and router schema.

Comment (devsergiy) [reply router/core/executor.go:103]: not fully true

In case schema has `inaccessible` directive there will be 2 schemas

Comment (Aenimus) [reply router/core/executor.go:103]: The client schema is a product of `@inaccessible` AND/OR `@tag` directives.

**Review by jensneuse — APPROVED** [review]

File: router/core/executor.go (hunk 1 above)
Comment (StarpTech) [reply router/core/executor.go:116]: Yes, but still it is nothing we want to get rid of in the nearly future.

**Pull request merged**